src/common/resolveEnvVariables.ts

```diff
@@ -0,0 +1,42 @@
+import fs from 'fs';
+
+type EnvObject = {
+  [variableName: string]: string;
+};
+
+//this function replaces all env variables within the docker compose yaml string with their values in the env file
+const resolveEnvVariables = (yamlText: string, filePath: string) => {
+  const envFileArray = filePath.split('/');
+  const envFilePath =
+    envFileArray.slice(0, envFileArray.length - 1).join('/') + '/.env';
```

Comment:
Better use https://nodejs.org/api/path.html#path_path_dirname_path and https://nodejs.org/api/path.html#path_path_join_paths for more portable / correct path handling